src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file radial_basis_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "dropout.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF(
+    const size_t inSize,
+    const size_t outSize) :
+    inSize(inSize),
+    outSize(outSize)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void RBF<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
+{
+  centres = arma::mat(outSize, input.n_cols, arma::fill::randu);
+  centres = arma::normcdf(centres, 0, 1);
+  sigmas = arma::ones(1,outSize);
+  arma::cube x = arma::cube(outSize, input.n_cols, input.n_rows);
```

Comment:
This might just be me, so I'll leave this comment up to you. But in cases like these, as a reader it helps me visualize it as stacking up `input` matrices into a cube. So instead it seems more intuitive to have `outSize`  to be the number of slices. I'll leave this up to your judgement, since this could be my personal idiosyncrasy :)

Reply:
Maybe I will try to do that but it will be quite difficult to calculate distances after doing this.

Reply:
@saksham189 What do you think of this??

Reply:
Hey @himanshupathak21061998 I went through the implementation here. Are you sure that the implementation is based on column-major matrices. 
(Matrices in mlpack are column-major, meaning that each column should correspond to a point in the dataset in contrast to row-major which is used by most libraries)

Reply:
No, right now it is in row-major format. I have to change this implementation to column-major it is not yet complete and giving the wrong predictions I want a piece of advice on what @favre49 is trying to suggest.

Reply:
I think there is no need to create the cube `x`. 
```
for each column in the input x
arma::mat temp = - center.each_col() -= x
```
Let me know what you think.

Reply:
Yeah, I like this idea will try to use it.